sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -52,14 +53,615 @@ def fix_big_data_passing(workflow: List[Dict[Text, Any]]) -> List[Dict[Text, Any
+        if (task_name, task_output.get('name')) in outputs_tasks:
+            if not task_spec.setdefault('workspaces', []):
+                task_spec['workspaces'].append({"name": task_name})
+            # Replace the args for the outputs in the task_spec
+            # $(results.task_output.get('name').path)  -->
+            # $(workspaces.task_name.path)/task_name-task_output.get('name')
+            placeholder = '$(results.%s.path)' % (task_output.get('name'))
+            workspaces_parameter = '$(workspaces.%s.path)/%s-%s' % (
+                task_name, task_name, task_output.get('name'))
+            task['spec'] = replace_bigger_data_placeholder(
+                task['spec'], placeholder, workspaces_parameter)
+
+    # Remove artifacts outputs from results
+    task['spec']['results'] = [
+        result for result in task_outputs
+        if (task_name, result.get('name')) not in outputs_tasks
+    ]
+
+    # Data passing for task inputs
+    task_spec = task.get('spec', {})
+    task_parmas = task_spec.get('params', [])
+    task_artifacts = task_spec.get('artifacts', [])
+    for task_parma in task_parmas:
+        if (task_name, task_parma.get('name')) in inputs_tasks:
+            if not task_spec.setdefault('workspaces', []):
+                task_spec['workspaces'].append({"name": task_name})
+            # Replace the args for the inputs in the task_spec
+            # /tmp/inputs/text/data ---->
+            # $(workspaces.task_name.path)/task_parma.get('name')
+            placeholder = '/tmp/inputs/text/data'
+            for task_artifact in task_artifacts:
+                if task_artifact.get('name') == task_parma.get('name'):
+                    placeholder = task_artifact.get('path')
+            workspaces_parameter = '$(workspaces.%s.path)/%s' % (
+                task_name, task_parma.get('name'))
+            task['spec'] = replace_bigger_data_placeholder(
+                task_spec, placeholder, workspaces_parameter)
+    # Handle the case of input artifact without dependent the output of other tasks
+    for task_artifact in task_artifacts:
+        if (task_name, task_artifact.get('name')) not in inputs_tasks:
+            # add input artifact processes
+            task = input_artifacts_tasks(task, task_artifact)
+
+    # Remove artifacts parameter from params
+    task['spec']['params'] = [
+        parma for parma in task_parmas
+        if (task_name, task_parma.get('name')) not in inputs_tasks
+    ]
+
+    # Remove artifacts from task_spec
+    if 'artifacts' in task_spec:
+        del task['spec']['artifacts']
+
+    return task
+
+
+# Create pvc for pipelinerun if bigger data passing.
+# As we used workspaces in tekton pipelines which depends on it.
+# User need to create PV manually, or enable dynamic volume provisioning, refer to the link of:
+# https://kubernetes.io/docs/concepts/storage/dynamic-provisioning
+# TODO: Remove PVC if Tekton version > = 0.12, use 'volumeClaimTemplate' instead
```

Comment:
The current design will use the same pvc for all the pipelinerun. Did Argo store them as "artifacts" in the original design?

Reply:
I difined different files for the bigger_data params,  I think we should reduce the pvc/pv number creating for one pipeline run, which mean reduce the effort for user of manual steps, there is no reason to create several pvcs for all the params.